backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/acl/service/AclManagerService.kt

```diff
@@ -0,0 +1,121 @@
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.security.acls.domain.ObjectIdentityImpl
+import org.springframework.security.acls.model.*
+import org.springframework.transaction.TransactionStatus
+import org.springframework.transaction.support.AbstractPlatformTransactionManager
+import org.springframework.transaction.support.TransactionCallback
+import java.io.Serializable
+import org.springframework.transaction.support.TransactionCallbackWithoutResult
+import org.springframework.transaction.support.TransactionTemplate
+import org.springframework.security.acls.model.MutableAcl
+import org.springframework.security.acls.model.ObjectIdentity
+import mu.KotlinLogging
+import org.springframework.security.acls.domain.BasePermission
+import java.lang.IllegalArgumentException
+
+private val logger = KotlinLogging.logger {  }
+
+interface AclManager {
+    fun <T> addPermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission)
+    fun <T> removePermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission)
+    fun <T> hasPermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission): Boolean
+    fun <T> clearPermissions(domainClass: Class<T>, id: Serializable)
+}
+
+@Service
+class SpecificationPermissionManager @Autowired constructor(private val aclManagerService : AclManagerService) {
+    fun addPermission(id: Long, sid: Sid, permission: Permission) {
+        logger.info("Add ${permissionToString(permission)} permission to spec $id ")
+        return aclManagerService.addPermission(Long::class.java, id, sid, permission)
+    }
+
+    fun removePermission(id: Long, sid: Sid, permission: Permission) {
+        logger.info("Remove ${permissionToString(permission)} permission from spec $id ")
+        return aclManagerService.removePermission(Long::class.java, id, sid, permission)
+    }
+
+    fun hasPermission(id: Long, sid: Sid, permission: Permission): Boolean {
+        val granted = aclManagerService.hasPermission(Long::class.java, id, sid, permission)
+        logger.info("$granted that $sid has ${permissionToString(permission)} permission on resource $id")
+        return granted
+    }
+
+    fun clearPermissions(id: Long) {
+        aclManagerService.clearPermissions(Long::class.java, id)
+    }
+
+    private fun permissionToString(permission: Permission): String = if (permission == BasePermission.READ) "READ" else "WRITE"
+}
+
+@Service
+class AclManagerService @Autowired constructor(private val aclService: MutableAclService,
+                                               private val transactionManager: AbstractPlatformTransactionManager) : AclManager {
+
+    override fun <T> addPermission(domainClass: Class<T>, id: Serializable, sid: Sid, permission: Permission) {
+        TransactionTemplate(transactionManager).execute(object : TransactionCallbackWithoutResult() {
```

Comment:
All this wrapping seems a bit weird. On the other hand, I have no experience with this stuff...